redundant-bmc/test/role_determination_test.cpp

```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: Apache-2.0
+#include "role_determination.hpp"
+
+#include <gtest/gtest.h>
+
+using Role =
```

Comment:
Noticed that this defined many places, would you prefer to move into `role_determination.hpp` and remove in other places?

Reply:
good idea, I'll look into the best place to define this once.

Reply:
I'm having a hard time on where there best place would be, since it's going to be used in other files that otherwise wouldn't need role_determination.hpp. 

Do you think a standalone header would make sense for this?  Like a types.hpp?  Otherwise, maybe I'll leave as-is for now.

Reply:
> Do you think a standalone header would make sense for this? Like a types.hpp?

Usually, I will follow this to define types/constants which could be used more than one place or some generic one.

> I'll leave as-is for now.

Sure,

Reply:
Waffled a bit on this, and then went with your first suggestion to put it role_determination.hpp.